src/encoder.rs

```diff
@@ -1162,6 +1164,82 @@ pub fn encode_tx_block<T: Pixel>(
     "mode.is_intra()={:#?}, plane={:#?}, tx_size.block_size()={:#?}, plane_bsize={:#?}, need_recon_pixel={:#?}",
     mode.is_intra(), p, tx_size.block_size(), plane_bsize, need_recon_pixel);
 
+  let ief_params =
+    if mode.is_directional() && fi.sequence.enable_intra_edge_filter {
```

Comment:
This should be a separate function I guess.

Reply:
Done in three functions, two for getting the surrounding block information, and one for `IntraEdgeFilterParameters` initialization.

Reply:
Neat :)